src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,126 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  onUploadedTodos: Todo[];
+  onCurrentTodos: (todos: Todo[]) => void;
+}
+
+enum Todos {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed ',
+}
+
+const filteredTodos = (todos: Todo[], filter: Todos) => {
+  switch (filter) {
+    case Todos.Active:
+      return todos.filter((todo) => !todo.completed);
+    case Todos.Completed:
+      return todos.filter((todo) => todo.completed);
+    default:
+      return todos;
+  }
+};
+
+const filteredTodoByQuery = (query: string, todos: Todo[]) => {
+  return todos.filter((todo) => {
+    const fixedTitle = todo.title.toLocaleLowerCase();
+    const fixedQuery = query.toLocaleLowerCase();
+
+    return fixedTitle.includes(fixedQuery);
+  });
+};
+
+const filteredTodosByInputAndStatus = (
```

Comment:
Consider renaming the 'filteredTodosByInputAndStatus' function to 'filterTodosByInputAndStatus' for better readability.